src/consensus/consensus.h

```diff
@@ -8,6 +8,12 @@
 
 #include <stdint.h>
 
+/**
+ * Maximum amount of time that a block timestamp is allowed to exceed the
+ * current network-adjusted time before the block will be accepted.
+ */
+static const int64_t MAX_FUTURE_BLOCK_TIME = 2 * 60 * 60;
```

Comment:
The goal is more to take it out of chain.h than to put it in consensus/consensus.h. consensus/header_verify would be fine as well. As an example, #8493 puts the check in libconsensus without putting chain.o in libconsensus too. Whether blocks eventually become valid when violating this rule or not, I think the question that should be asked is "in a hypothetical finished libconsensus, is this check included or not?", and I also think the answer is yes.